File: InterviewTest/Models/HeroModel.cs
using System.Collections.Generic;

namespace InterviewTest.Models
{
    public class HeroModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Power { get; set; }
        public List<KeyValuePair<string, int>> Stats { get; set; }
    }
}

File: FrontEnd/src/app/Hero/hero.model.ts
import { StaticSymbolCache } from "@angular/compiler";

export interface KeyValuePair{
    key:string;
    value:number;
}

export class Hero{
    public lastUpdated: boolean;
    public id?:string;
    public name:string;
    public power:string;
    public stats: KeyValuePair[];
    constructor( heroName: string, heroPower:string, stats: KeyValuePair[])
    {
        this.name = heroName;
        this.power = heroPower;
        this.stats = stats;
    }    
}

